modules/recon/masscan/masscan.go

```diff
@@ -0,0 +1,347 @@
+	"io/ioutil"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"os"
+	"path/filepath"
+
+	"github.com/netm4ul/netm4ul/core/database/models"
+
+	"github.com/BurntSushi/toml"
+	"github.com/netm4ul/netm4ul/modules"
+)
+
+// MasscanResult represent the parsed ouput
+type MasscanResult struct {
+	Resultat []Scan
+}
+
+// Scan represents the ip and ports output
+type Scan struct {
+	IP        string  `json:"ip"`
+	Timestamp string  `json:"timestamp"`
+	Ports     []Ports `json:"ports"`
+}
+
+// Ports represents the port, proto, service, ttl, reason and status output
+type Ports struct {
+	Port    uint16  `json:"port"`
+	Proto   string  `json:"proto"`
+	Service Service `json:"service,omitempty"`
+	TTL     int     `json:"ttl"`
+	Reason  string  `json:"reason"`
+	Status  string  `json:"status"`
+}
+
+// Service represents the name and the banner output
+type Service struct {
+	Name   string `json:"name"`
+	Banner string `json:"banner"`
+}
+
+// ConfigToml : configuration model (from the toml file)
+type ConfigToml struct {
+	Verbose           bool   `toml:"verbose"`
+	VeryVerbose       bool   `toml:"very-verbose"`
+	Rate              int    `toml:"rate"`
+	Ping              bool   `toml:"ping"`
+	Seed              int    `toml:"seed"`
+	Adapter           string `toml:"adapter"`
+	AdapterIP         string `toml:"adapter-ip"`
+	AdapterMAC        string `toml:"adapter-mac"`
+	AdapterVLAN       string `toml:"adapter-vlan"`
+	RouterMAC         string `toml:"router-mac"`
+	Retries           int    `toml:"retries"`
+	MinPacket         int    `toml:"min-packet"`
+	HTTPUserAgent     string `toml:"http-user-agent"`
+	RandomizeHosts    bool   `toml:"randomize-hosts"`
+	Exclude           string `toml:"exclude"`
+	Banners           bool   `toml:"banners"`
+	Ports             string `toml:"ports"`
+	ConnectionTimeout int    `toml:"connection-timeout"`
+	SourcePort        int    `toml:"source-port"`
+	TTL               int    `toml:"ttl"`
+	Wait              string `toml:"wait"`
+}
+
+// Masscan "class"
+type Masscan struct {
+	// Config : exported config
+	Config ConfigToml
+}
+
+//NewMasscan generate a new Masscan module (type modules.Module)
+func NewMasscan() modules.Module {
+	gob.Register(MasscanResult{})
+	var m modules.Module
+	m = &Masscan{}
+	return m
+}
+
+// Name : name getter
+func (M *Masscan) Name() string {
+	return "Masscan"
+}
+
+// Author : Author getter
+func (M *Masscan) Author() string {
+	return "soldat-ryan"
+}
+
+// Version : Version  getter
+func (M *Masscan) Version() string {
+	return "0.9"
+}
+
+// DependsOn : Generate the dependencies requirement
+func (M *Masscan) DependsOn() []modules.Condition {
+	var _ modules.Condition
+	return []modules.Condition{}
+}
+
+// Run : Main function of the module
+func (M *Masscan) Run(inputs []modules.Input) (modules.Result, error) {
+	log.Debug("H3ll-0 M4sscan")
+
+	// Temporary IP forced : 212.47.247.190 = edznux.fr
+	outputfile := "/tmp/" + generateUUID() + ".json"
```

Comment:
we might prefer https://golang.org/pkg/io/ioutil/#TempFile but that's a details (masscan is *nix only soo..)